frontend-react/src/pages/admin/AdminRevHistory.tsx

```diff
@@ -0,0 +1,273 @@
+ * the org and settingType cgi params */
+interface MainComponentProps extends SettingRevisionParams {
+    leftSelectedListItem: string;
+    rightSelectedListItem: string;
+    onClickHandler: (
+        key: string,
+        itemClickedKey: string,
+        data: SettingRevision
+    ) => void;
+}
+
+/**
+ * Nest loading into a component that can spin so the bulk of the page loads while
+ * the network request happens.
+ */
+const MainRevHistoryComponent = (props: MainComponentProps) => {
+    const { data, isLoading, isError } =
+        useSettingRevisionEndpointsQuery(props);
+    const msg = isError
+        ? "Failed to load data"
+        : isLoading
+        ? "Loading..."
+        : "Data not found"; // should not be used because `!data` test below but useful for unit test debugging
+    return (
+        <Grid col={"fill"} className={"rs-maxwidth-vw80"}>
+            <Grid row gap="md" className={"rs-accord-list-row"}>
+                <Grid className={"rs-list-diffs-items"}>
+                    {!data ? (
+                        <div>{msg}</div>
+                    ) : (
+                        <Accordion
+                            bordered={false}
+                            items={dataToAccordionItems({
+                                key: "left",
+                                selectedKey: props.leftSelectedListItem,
+                                onClickHandler: props.onClickHandler,
+                                data,
+                            })}
+                        />
+                    )}
+                </Grid>
+                <Grid className={"rs-list-diffs-items"}>
+                    {!data ? null : (
+                        <Accordion
+                            bordered={false}
+                            items={dataToAccordionItems({
+                                key: "right",
+                                selectedKey: props.rightSelectedListItem,
+                                onClickHandler: props.onClickHandler,
+                                data,
+                            })}
+                        />
+                    )}
+                </Grid>
+            </Grid>
+        </Grid>
+    );
+};
+
+/** main page, not exported here because it should only be loaded via AdminRevHistoryWithAuth() **/
+const AdminRevHistory = () => {
+    const { org, settingType } = useParams<SettingRevisionParams>(); // props past to page via the route/url path args
+    const [leftJson, setLeftJson] = useState("");
+    const [rightJson, setRightJson] = useState("");
+    // used to highlight which item is selected.
+    const [leftSelectedListItem, setLeftSelectedListItem] = useState("");
+    const [rightSelectedListItem, setRightSelectedListItem] = useState("");
+    const [leftItem, setLeftItem] = useState<SettingRevision | null>(null);
+    const [rightItem, setRightItem] = useState<SettingRevision | null>(null);
```

Comment:
great candidate for a useReducer...

Reply:
Agreed. But can I do that in another PR so to get this one merged it?

Reply:
Yup!